osquery/extensions/extensions.cpp

```diff
@@ -89,17 +91,30 @@ void ExtensionWatcher::exitFatal(int return_code) {
 }
 
 void ExtensionWatcher::watch() {
+  // Attempt to ping the extension core.
+  // This does NOT use pingExtension to avoid the latency checks applied.
   ExtensionStatus status;
-  try {
-    auto client = EXManagerClient(path_);
-    // Ping the extension manager until it goes down.
-    client.get()->ping(status);
-  } catch (const std::exception& e) {
-    LOG(WARNING) << "Extension watcher ending: osquery core has gone away";
+  bool core_sane = true;
+  if (isWritable(path_)) {
```

Comment:
This check helps prevent spurious warnings in older version of Thrift. There is still a tear-down race within exceptions when aborting their thread pool in response to a killed core/watcher. Solving this race is not high priority at the moment.